system-x/services/tekton/src/main/java/software/tnb/tekton/service/Tekton.java

```diff
@@ -0,0 +1,120 @@
+package software.tnb.tekton.service;
+
+import software.tnb.common.account.NoAccount;
+import software.tnb.common.deployment.OpenshiftDeployable;
+import software.tnb.common.deployment.WithOperatorHub;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.service.Service;
+import software.tnb.common.utils.WaitUtils;
+import software.tnb.tekton.validation.TektonValidation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+
+import java.io.IOException;
+import java.util.function.Predicate;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.tekton.client.TektonClient;
+
+@AutoService(Tekton.class)
+public class Tekton extends Service<NoAccount, TektonClient, TektonValidation> implements OpenshiftDeployable, WithOperatorHub {
+    private static final Logger LOG = LoggerFactory.getLogger(Tekton.class);
+
+    public Tekton() {
+
+    }
+
+    protected TektonClient client() {
+        if (OpenshiftClient.get().isAdaptable(TektonClient.class)) {
+            return OpenshiftClient.get().adapt(TektonClient.class);
+        }
+        throw new IllegalArgumentException("Unable to adapt OpenshiftClient to TektonClient");
+    }
+
+    @Override
+    public TektonValidation validation() {
+        if (validation == null) {
+            validation = new TektonValidation(client());
+        }
+        return validation;
+    }
+
+    public void create() {
+        LOG.debug("Creating Tekton operator");
+        createSubscription();
+    }
+
+    public void undeploy() {
+        LOG.debug("Undeploy Tekton operator");
```

Comment:
since it is clusterwide, maybe we could leave it installed? I assume it will affect running tests when you undeploy tekton in the middle of some other test (the probability is maybe low, but you never know)